extensions/youtrack/src/work-item.tsx

```diff
@@ -0,0 +1,88 @@
+import { IssueExtended, WorkItemSubmit } from "./interfaces";
+import { WorkItem } from "youtrack-rest-client";
+import { useEffect, useState } from "react";
+import { Action, ActionPanel, Detail, Form, Icon, showToast, Toast, useNavigation } from "@raycast/api";
+
+export function AddWork(props: {
+  getIssueDetailsCb: () => Promise<IssueExtended> | null;
+  createWorkItemCb: (workItem: WorkItem) => Promise<WorkItem> | null;
+  link: string;
+  instance: string;
+}) {
+  const [issue, setIssue] = useState<IssueExtended | null>(null);
+  const [submitted, setSubmitted] = useState(false);
+  const { pop } = useNavigation();
+  useEffect(() => {
+    async function fetchIssueDetails() {
+      const issue = await props.getIssueDetailsCb();
+      setIssue(issue);
+    }
+
+    fetchIssueDetails();
+  }, []);
+
+  if (!issue) {
+    return <Detail isLoading />;
+  }
+  const workTypes = issue.workItemTypes?.length ? issue.workItemTypes : [];
+  if (!workTypes.find((workType) => workType.id === "")) {
+    workTypes.push({ id: "", name: "- without -" });
+  }
+
+  const handleSubmit = async (values: WorkItemSubmit) => {
+    if (!props.createWorkItemCb) {
+      return;
+    }
+    const workItem: WorkItem = {
+      text: values.comment,
+      date: values.date.valueOf(),
+      duration: {
+        presentation: values.time,
+      },
+    };
+    try {
+      await props.createWorkItemCb(workItem);
+      setSubmitted(true);
+      showToast({
+        style: Toast.Style.Success,
+        title: "Work item added",
+      });
+    } catch (error) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Failed adding work item",
+        // @ts-expect-error message is not defined on Error
+        message: `${error?.message || "Unknown error occurred"}`,
+      });
+      return;
+    }
+  };
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          {submitted ? (
+            <Action icon={Icon.ArrowLeft} title="Go Back" onAction={pop} />
```

Comment:
Not sure this work, maybe you can show a toast instead and pop?

Reply:
Yes, this works.
Using the toast is also a great idea. I implemented that as an additional way to return to the previous view. My first implementation offers a seamless workflow because the user can use the same command as he used to commit the item right before. In this way, the user can create the work items and go back very fast. What do you think about my solution?

Reply:
We don't use pop in other view of Raycast so it feels a bit weird, I would rather use the toast. I don't use youTrack myself, could you record a video of it so I can see it in action?

Reply:
I have been working with the extension for the last two weeks. You are right. Using the toast or the escape key is more intuitive than using the pop to switch to the previous view.